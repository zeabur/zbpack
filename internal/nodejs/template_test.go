package nodejs_test

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
	"github.com/zeabur/zbpack/internal/nodejs"
)

func TestTemplate_NBuildCmd_NOutputDir(t *testing.T) {
	ctx := nodejs.TemplateContext{
		NodeVersion: "18",

		InstallCmd: "RUN yarn install",
		BuildCmd:   "",
		StartCmd:   "yarn start",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestTemplate_NBuildCmd_OutputDir_NSPA(t *testing.T) {
	ctx := nodejs.TemplateContext{
		NodeVersion: "18",

		InstallCmd: "RUN yarn install",
		BuildCmd:   "",
		StartCmd:   "yarn start",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestTemplate_NBuildCmd_OutputDir_SPA(t *testing.T) {
	ctx := nodejs.TemplateContext{
		NodeVersion: "18",

		InstallCmd: "RUN yarn install",
		BuildCmd:   "",
		StartCmd:   "yarn start",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestTemplate_BuildCmd_NOutputDir(t *testing.T) {
	ctx := nodejs.TemplateContext{
		NodeVersion: "18",

		InstallCmd: "RUN yarn install",
		BuildCmd:   "yarn build",
		StartCmd:   "yarn start",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestTemplate_BuildCmd_OutputDir(t *testing.T) {
	ctx := nodejs.TemplateContext{
		NodeVersion: "18",

		InstallCmd: "RUN yarn install",
		BuildCmd:   "yarn build",
		StartCmd:   "yarn start",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}

func TestTemplate_BuildCmd_Bun(t *testing.T) {
	ctx := nodejs.TemplateContext{
		Bun:         true,
		NodeVersion: "18",
		InstallCmd:  "RUN bun install",
		StartCmd:    "bun start main.ts",
	}

	result, err := ctx.Execute()
	assert.NoError(t, err)
	snaps.MatchSnapshot(t, result)
}
