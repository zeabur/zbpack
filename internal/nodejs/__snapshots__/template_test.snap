
[TestTemplate_NBuildCmd_NOutputDir - 1]
FROM node:18 as build

ENV PORT=8080
WORKDIR /src

RUN corepack enable && corepack prepare --all
COPY . .

RUN yarn install

# Build if we can build it



EXPOSE 8080
CMD yarn start


---

[TestTemplate_NBuildCmd_OutputDir_NSPA - 1]
FROM node:18 as build

ENV PORT=8080
WORKDIR /src

RUN corepack enable && corepack prepare --all
COPY . .

RUN yarn install

# Build if we can build it



FROM nginx:alpine as runtime

COPY --from=build /src/dist /usr/share/nginx/html/static
RUN echo "\
    server { \
        listen 8080; \
        root /usr/share/nginx/html/static; \
        absolute_redirect off; \
        location / { \
            try_files \$uri \$uri.html \$uri/index.html /404.html =404; \
        } \
    }"> /etc/nginx/conf.d/default.conf
EXPOSE 8080


---

[TestTemplate_NBuildCmd_OutputDir_SPA - 1]
FROM node:18 as build

ENV PORT=8080
WORKDIR /src

RUN corepack enable && corepack prepare --all
COPY . .

RUN yarn install

# Build if we can build it



FROM nginx:alpine as runtime

COPY --from=build /src/dist /usr/share/nginx/html/static
RUN echo "\
    server { \
        listen 8080; \
        root /usr/share/nginx/html/static; \
        absolute_redirect off; \
        location / { \
            try_files \$uri /index.html; \
        } \
    }"> /etc/nginx/conf.d/default.conf
EXPOSE 8080


---

[TestTemplate_BuildCmd_NOutputDir - 1]
FROM node:18 as build

ENV PORT=8080
WORKDIR /src

RUN corepack enable && corepack prepare --all
COPY . .

RUN yarn install

# Build if we can build it
RUN yarn build


EXPOSE 8080
CMD yarn start


---

[TestTemplate_BuildCmd_OutputDir - 1]
FROM node:18 as build

ENV PORT=8080
WORKDIR /src

RUN corepack enable && corepack prepare --all
COPY . .

RUN yarn install

# Build if we can build it
RUN yarn build


FROM nginx:alpine as runtime

COPY --from=build /src/dist /usr/share/nginx/html/static
RUN echo "\
    server { \
        listen 8080; \
        root /usr/share/nginx/html/static; \
        absolute_redirect off; \
        location / { \
            try_files \$uri /index.html; \
        } \
    }"> /etc/nginx/conf.d/default.conf
EXPOSE 8080


---

[TestTemplate_BuildCmd_Bun - 1]
# Install bun if we need it
FROM oven/bun:1.0 as bun-runtime
FROM node:18 as build

ENV PORT=8080
WORKDIR /src
# Copy the bun binary from the bun-runtime stage directly.
# A bit hacky but it works.
COPY --from=bun-runtime /usr/local/bin/bun /usr/local/bin
COPY --from=bun-runtime /usr/local/bin/bunx /usr/local/bin

RUN corepack enable && corepack prepare --all
COPY . .

RUN bun install

# Build if we can build it



EXPOSE 8080
CMD bun start main.ts


---
